File: java/reverse-string/src/main/java/ReverseString.java
class ReverseString {

    String reverse(String inputString) {
        String temp ="";
        if (inputString.isEmpty())
            return temp;
        else {
            for(int i = inputString.length() - 1; i >= 0; i--){
                temp = temp + inputString.charAt(i);
            }
            return temp;
        }

    }
  
}

File: javascript/reverse-string/reverse-string.js
var reverseString = function (input) {
  this.value = input;
  var response;

  if   (0 === this.value.length) {
    response = '';
  } else {
    var temp = '';
    var valueArray = this.value.toString().split("");
    for (i = valueArray.length -1; i >= 0; i--) {
      temp = temp + valueArray[i];
      response = temp;
    }
  }

  return response;
};

module.exports = reverseString;
